src/lib/structures/ExtendedArgument.ts

```diff
@@ -80,7 +80,7 @@ export abstract class ExtendedArgument<K extends keyof ArgType, T> extends Argum
 		// If the result was successful (i.e. is of type `Ok<ArgType[K]>`), pass its
 		// value to [[ExtendedArgument#handle]] for further parsing. Otherwise, return
 		// the error as is; it'll provide contextual information from the base argument.
-		return isOk(result) ? this.handle(result.value, { ...context, parameter }) : result;
+		return result.isOk() ? this.handle(result.value, { ...context, parameter }) : result;
```

Comment:
This doesn't compile

Reply:
That's because it needs `result.isOkAnd((value) => this.handle(value, { ...context, parameter })`. Could be tricky if `this.handle` returns a Promise, though, if that's the case, `result.unwrap()` always works.

Reply:
ah that's what I was looking for. Ty.

Reply:
doesn't matter now the class is removed though ;)